presto-session-property-managers/src/main/java/io/prestosql/plugin/session/file/FileSessionPropertyManager.java

```diff
@@ -15,36 +15,33 @@
 
 import com.fasterxml.jackson.databind.JsonMappingException;
 import com.fasterxml.jackson.databind.exc.UnrecognizedPropertyException;
-import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableList;
 import io.airlift.json.JsonCodec;
 import io.airlift.json.JsonCodecFactory;
 import io.airlift.json.ObjectMapperProvider;
+import io.prestosql.plugin.session.AbstractSessionPropertyManager;
 import io.prestosql.plugin.session.SessionMatchSpec;
-import io.prestosql.spi.session.SessionConfigurationContext;
-import io.prestosql.spi.session.SessionPropertyConfigurationManager;
 
 import javax.inject.Inject;
 
 import java.io.IOException;
 import java.io.UncheckedIOException;
 import java.nio.file.Files;
 import java.nio.file.Path;
-import java.util.HashMap;
 import java.util.List;
-import java.util.Map;
 
 import static com.fasterxml.jackson.databind.DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES;
 import static java.lang.String.format;
 import static java.util.Objects.requireNonNull;
 
 public class FileSessionPropertyManager
-        implements SessionPropertyConfigurationManager
+        extends AbstractSessionPropertyManager
 {
     public static final JsonCodec<List<SessionMatchSpec>> CODEC = new JsonCodecFactory(
             () -> new ObjectMapperProvider().get().enable(FAIL_ON_UNKNOWN_PROPERTIES))
             .listJsonCodec(SessionMatchSpec.class);
 
-    private final List<SessionMatchSpec> sessionMatchSpecs;
+    private final ImmutableList<SessionMatchSpec> sessionMatchSpecs;
```

Comment:
We could just use interface type in the declaration

Reply:
Made it explicit in the type so we can rely on it in getSessionMatchSpecs() and return the list without doing a ImmutableList.copyOf().

Reply:
Made it explicit in the type so we can rely on it in getSessionMatchSpecs() and return the list without doing a ImmutableList.copyOf().